tests/pytests/functional/states/test_reg.py

```diff
@@ -0,0 +1,342 @@
+import pytest
+import salt.states.reg as reg
+import salt.utils.win_dacl as win_dacl
+import salt.utils.win_functions as win_functions
+import salt.utils.win_reg as reg_util
+from tests.support.mock import patch
+
+try:
+    CURRENT_USER = win_functions.get_current_user(with_domain=False)
+except NameError:
+    # Not a Windows Machine
+    pass
+
+HIVE = "HKEY_CURRENT_USER"
```

Comment:
We want to try to avoid using module level variables here.

Reply:
Created a class for these tests